src/plugins/workspace/server/saved_objects/saved_objects_wrapper_for_check_workspace_conflict.ts

```diff
@@ -0,0 +1,257 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import Boom from '@hapi/boom';
+import {
+  SavedObject,
+  SavedObjectsBaseOptions,
+  SavedObjectsBulkCreateObject,
+  SavedObjectsBulkResponse,
+  SavedObjectsClientWrapperFactory,
+  SavedObjectsCreateOptions,
+  SavedObjectsErrorHelpers,
+  SavedObjectsUtils,
+  SavedObjectsSerializer,
+  SavedObjectsCheckConflictsObject,
+  SavedObjectsCheckConflictsResponse,
+} from '../../../../core/server';
+
+const errorContent = (error: Boom.Boom) => error.output.payload;
+
+export class WorkspaceConflictSavedObjectsClientWrapper {
+  private _serializer?: SavedObjectsSerializer;
+  public setSerializer(serializer: SavedObjectsSerializer) {
+    this._serializer = serializer;
+  }
+  private getRawId(props: { namespace?: string; id: string; type: string }) {
+    return (
+      this._serializer?.generateRawId(props.namespace, props.type, props.id) ||
+      `${props.type}:${props.id}`
+    );
+  }
+  public wrapperFactory: SavedObjectsClientWrapperFactory = (wrapperOptions) => {
+    const createWithWorkspaceConflictCheck = async <T = unknown>(
+      type: string,
+      attributes: T,
+      options: SavedObjectsCreateOptions = {}
+    ) => {
+      const { workspaces, id, overwrite } = options;
+      let savedObjectWorkspaces = options?.workspaces;
+
+      if (id && overwrite && workspaces) {
+        let currentItem;
+        try {
+          currentItem = await wrapperOptions.client.get(type, id);
+        } catch (e) {
+          // this.get will throw an error when no items can be found
+        }
+        if (currentItem) {
+          if (
+            SavedObjectsUtils.filterWorkspacesAccordingToSourceWorkspaces(
+              workspaces,
+              currentItem.workspaces
+            ).length
+          ) {
+            throw SavedObjectsErrorHelpers.createConflictError(type, id);
+          } else {
+            savedObjectWorkspaces = currentItem.workspaces;
+          }
+        }
+      }
+
+      return await wrapperOptions.client.create(type, attributes, {
+        ...options,
+        workspaces: savedObjectWorkspaces,
+      });
+    };
+
+    const bulkCreateWithWorkspaceConflictCheck = async <T = unknown>(
+      objects: Array<SavedObjectsBulkCreateObject<T>>,
+      options: SavedObjectsCreateOptions = {}
+    ): Promise<SavedObjectsBulkResponse<T>> => {
+      const { overwrite, namespace } = options;
```

Comment:
Shall we check the override option first? Then we don't need to get docs and check workspace conflicts here.

Reply:
You are right, if overwrite is false, I will make bulkGetDocs to [].